src/RTC_PCF8523.cpp

```diff
@@ -1,292 +1,334 @@
+      - CTBF: Timer B flag indicates that interrupt was triggered. Though
+        typically used for non-pulsed mode, user may wish to query this later.
+*/
+/**************************************************************************/
+void RTC_PCF8523::disableCountdownTimer() {
+  // TBC disable to stop Timer B clock
+  write_register(PCF8523_CLKOUTCONTROL,
+                 ~1 & read_register(PCF8523_CLKOUTCONTROL));
+}
+
+/**************************************************************************/
+/*!
+    @brief  Stop all timers, clear their flags and settings on the PCF8523.
+    @details This includes the Countdown Timer, Second Timer, and any CLKOUT
+   square wave configured with writeSqwPinMode().
+*/
+/**************************************************************************/
+void RTC_PCF8523::deconfigureAllTimers() {
+  disableSecondTimer(); // Surgically clears CONTROL_1
+  write_register(PCF8523_CONTROL_2, 0);
+  write_register(PCF8523_CLKOUTCONTROL, 0);
+  write_register(PCF8523_TIMER_B_FRCTL, 0);
+  write_register(PCF8523_TIMER_B_VALUE, 0);
+}
+
+/**************************************************************************/
+/*!
+    @brief Compensate the drift of the RTC.
+    @details This method sets the "offset" register of the PCF8523,
+      which can be used to correct a previously measured drift rate.
+      Two correction modes are available:
+
+      - **PCF8523\_TwoHours**: Clock adjustments are performed on
+        `offset` consecutive minutes every two hours. This is the most
+        energy-efficient mode.
+
+      - **PCF8523\_OneMinute**: Clock adjustments are performed on
+        `offset` consecutive seconds every minute. Extra adjustments are
+        performed on the last second of the minute is `abs(offset)>60`.
+
+      The `offset` parameter sets the correction amount in units of
+      roughly 4&nbsp;ppm. The exact unit depends on the selected mode:
+
+      |  mode               | offset unit                            |
+      |---------------------|----------------------------------------|
+      | `PCF8523_TwoHours`  | 4.340 ppm = 0.375 s/day = 2.625 s/week |
+      | `PCF8523_OneMinute` | 4.069 ppm = 0.352 s/day = 2.461 s/week |
+
+      See the accompanying sketch pcf8523.ino for an example on how to
+      use this method.
+
+    @param mode Correction mode, either `PCF8523_TwoHours` or
+      `PCF8523_OneMinute`.
+    @param offset Correction amount, from -64 to +63. A positive offset
+      makes the clock slower.
+*/
+/**************************************************************************/
+void RTC_PCF8523::calibrate(Pcf8523OffsetMode mode, int8_t offset) {
+  write_register(PCF8523_OFFSET, ((uint8_t)offset & 0x7F) | mode);
+}
+
+/**************************************************************************/
+/*!
+    @brief read the offset register 
+*/
+/**************************************************************************/
+
+int8_t RTC_PCF8523::readOffsetReg() {

```

Comment:
Return type should be `uint8_t`: it is just a byte; it does not make sense to interpret it as a signed number.

What is the purpose of this method? The way the data is encoded within the offset register of the RTC is an implementation detail the user should not need to be aware of.

The presence of this method could be justified as a way to read both the offset and the mode at once, provided suitable methods are provided for decoding it, e.g.

```c++
static Pcf8523OffsetMode getOffsetMode(uint8_t offset_reg);
static int8_t getOffsetValue(uint8_t offset_reg);
```

In which case the next two methods could be dropped.

Reply:
Yes, I see your point and you are quite correct. The byte does not represent a signed interger. 
I will see if I can also work it out with your suggestion.
Thanks

Reply:
@edgar-bonet 
I hope you don't mind me continueing this conversation.
I made some changes by directly editing in Github and that generated a whole lot of PR runs for every time I committed a change. This in turn created a lot of apparent failures.
the latest is:
`Run bash ci/doxy_gen_and_deploy.sh src
Setting up the script...
Cloning into 'RTClib'...
Generating Doxygen code documentation...
Grabbing default Doxyfile
/home/runner/work/RTClib/RTClib/src/RTClib.h:426: warning: return type of member RTC_PCF8523::readOffsetReg is not documented
/home/runner/work/RTClib/RTClib/src/RTClib.h:427: warning: return type of member RTC_PCF8523::getOffsetMode is not documented
/home/runner/work/RTClib/RTClib/src/RTClib.h:428: warning: return type of member RTC_PCF8523::getOffset is not documented
Error: Process completed with exit code 1.`
I notice in the suggested code above that you used 'static' as far as I can see this only means that once used in a function the variable value is recorded for possible future use in the same funtion but can not be used elsewhere.
I am not sure whether using static in this mode means that the return value is then able to be used in the call to the function from the example sketch.
The reason that I used String is that I was trying to make the method as transparent as possible to the user by returning the actual text. In my original version (only local) I used a global variable but I realised this was not a good idea,
If I return a value, 0 or 1, or a bool, it would still need interpretation in the sketch (as in Method 1 in the sketch)instead of in the library.
I made two possible methods in the example sketch with the idea that whoever eventually merges the branch can choose to leave both or only one of the methods and edit the code as appropriate.
My motivation was more to put forward a concept rather than, perhaps, the final solution. but in order to carry this forward I am quite happy to be guided by you.
My experience with the PCF8523 RTC is that calibration is invariably required so I think this is an important issue.

Reply:
I am replying in the main thread.